tests/happy_path/test_node-operators-registry-happy-path.py

```diff
@@ -0,0 +1,568 @@
+import pytest
+from web3 import Web3
+from brownie import chain, ZERO_ADDRESS
+from hexbytes import HexBytes
+from typing import NewType, Tuple
+
+from utils.test.extra_data import (
+    ExtraDataService,
+)
+from utils.test.helpers import (
+    shares_balance,
+    ETH,
+    almostEq
+)
+from utils.test.oracle_report_helpers import (
+    oracle_report,
+)
+from utils.config import (contracts, lido_dao_staking_router)
+
+PUBKEY_LENGTH = 48
+SIGNATURE_LENGTH = 96
+INITIAL_TOKEN_HOLDER = "0x000000000000000000000000000000000000dead"
+TOTAL_BASIS_POINTS = 10000
+ZERO_HASH = bytes([0] * 32)
+ZERO_BYTES32 = HexBytes(ZERO_HASH)
+
+
+@pytest.fixture()
+def extra_data_service():
+    return ExtraDataService()
+
+
+@pytest.fixture(scope="module")
+def voting_eoa(accounts):
+    return accounts.at(contracts.voting.address, force=True)
+
+
+StakingModuleId = NewType('StakingModuleId', int)
+NodeOperatorId = NewType('NodeOperatorId', int)
+NodeOperatorGlobalIndex = Tuple[StakingModuleId, NodeOperatorId]
+
+
+def node_operator(module_id, node_operator_id) -> NodeOperatorGlobalIndex:
+    return module_id, node_operator_id
+
+
+@pytest.fixture(scope="module")
+def nor(interface):
+    return interface.NodeOperatorsRegistry(contracts.node_operators_registry.address)
+
+
+def calc_no_rewards(
+    nor, no_id, report_shares
+):
+    total_signing_keys = nor.getNodeOperatorSummary(no_id)[
+        'totalDepositedValidators'] - nor.getNodeOperatorSummary(no_id)['totalExitedValidators']
+    total_deposite_keys = nor.getStakingModuleSummary()[
+        'totalDepositedValidators'] - nor.getStakingModuleSummary()['totalExitedValidators']
+
+    return report_shares // 2 * total_signing_keys // total_deposite_keys
+
+
+def increase_limit(nor, first, second, base, keys_count, voting_eoa):
```

Comment:
Didn't get the idea of calculating `max(vetted, total_added)`

Total added ≥ vetted always:
https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/contracts/0.4.24/nos/NodeOperatorsRegistry.sol#L168-L185

Should rather calculate the new limit (i.e., new vetted) as follows:
```new_vetted = prev_vetted + keys_count - exited```